MidCamp/midcamp
# Virtual Room Links

# Description

- Adds a link field to the Location taxonomy term
- Configures default view mode for Location taxonomy to simply display the name of the term
- Creates custom view mode for Location taxonomy term to display the link provided for the term
- Targets Topic content tagged with the 2020 event year to display the custom view mode so prior years (and possibly future years) are not impacted
- Adds template suggestions for taxonomy terms to include bundle and view mode

What all of the above amounts to is a link on 2020 Topic nodes to allow users to enter the conference room, while other years should remain unchanged.

# To test

- Log in to the site at https://nginx-midcamp-org-feature-room-links.us.amazee.io/user/login
- Look at some of the Location terms for the rooms we are using this year (eg https://nginx-midcamp-org-feature-room-links.us.amazee.io/taxonomy/term/81/edit)
- Observe there is a link field there for adding links to Zoom, Google Hangouts, whatever
- Browse the 2020 schedule at https://nginx-midcamp-org-feature-room-links.us.amazee.io/2020/schedule/friday, clicking on various sessions
- Observe the room name has been replaced by the link set on the term, with language like "To to Room 314"

# Screenshots

## Taxonomy Term

![image](https://user-images.githubusercontent.com/4048700/76709387-51cd6c00-66cc-11ea-9b46-0ffa2cf70410.png)

## 2020 Event Sessions (Topic nodes)

![image](https://user-images.githubusercontent.com/4048700/76709320-ef746b80-66cb-11ea-9e46-1d2b98fc18c6.png)

## 2019 Event Sessions (Topic nodes)

![image](https://user-images.githubusercontent.com/4048700/76709342-19c62900-66cc-11ea-8950-030269cc8ba8.png)

# Outstanding Questions

- [ ] Do we want to integrate this into the Schedule View? 
- [ ] How can we promote the virtual rooms best on the homepage?
